frontends/web/src/hooks/api.test.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+import { SubscriptionCallback } from '../api/subscribe';
+import { useSubscribe, useLoad, useSync } from './api';
+import * as utils from './utils';
+import { TStatus } from '../api/coins';
+
+const useMountedRefSpy = jest.spyOn(utils, 'useMountedRef');
+
+describe('hooks for api calls', () => {
+
+  beforeEach(() => {
+    useMountedRefSpy.mockReturnValue({ current: true });
+  });
+
+  it('useSubscribe should return proper value of a subscription function', () => {
+    const MOCK_RETURN_STATUS: TStatus = {
+      tipAtInitTime: 2408855,
+      tip: 2408940,
+      tipHashHex: '0000000000000015f61742c773181dd368527575a6ac02ea5ecbace8e73cc083',
+      targetHeight: 2408940
+    };
+
+    const mockSubscribe = jest.fn().mockImplementation(() => (cb: SubscriptionCallback<any>) => mockSubscribeEndpoint(cb));
+    const mockSubscribeEndpoint = jest.fn().mockImplementation((cb) => cb(MOCK_RETURN_STATUS));
+
+
+    const { result } = renderHook(() => useSubscribe(mockSubscribe()));
+
+    expect(result.current).toBe(MOCK_RETURN_STATUS);
+  });
+
+  it('useLoad should load promise and return the correct resolved value', async () => {
+    const mockApiCall = jest.fn().mockImplementation(() => () => Promise.resolve(true));
+    const { result, waitForNextUpdate } = renderHook(() => useLoad(mockApiCall()));
+    await waitForNextUpdate();
+    expect(result.current).toBe(true);
+  });
```

Comment:
@shonsirsha if possible, could you add another test for `useLoad` that tests the 2nd argument as well?

I've added the 2nd arg a while ago (not 100% sure if this was a good idea), as useLoad only does 1 request on mount,  
see my comment here for an example: https://github.com/digitalbitbox/bitbox-wallet-app/pull/1851/files#r1039794716

Reply:
Yes sure, I'll write a test for the 2nd arg. In this test, I'll assert that the apiCall should be called however many times the dependencies (2nd arg) changes.  🙂